modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/IgniteH2Indexing.java

```diff
@@ -2906,7 +2908,12 @@ private UpdateResult executeUpdateNonTransactional(
             .setLocal(qryDesc.local())
             .setPageSize(qryParams.pageSize())
             .setTimeout(qryParams.timeout(), TimeUnit.MILLISECONDS)
-            .setMaxMemory(qryParams.maxMemory());
+            .setMaxMemory(qryParams.maxMemory())
+            // On no MVCC mode we cannot use lazy mode when UPDATE query contains updated columns
+            // in WHERE condition because it may be cause pf update one entry several times
+            // (when index for such columns is selected for scan):
+            // e.g. : UPDATE test SET val = val + 1 WHERE val >= ?
+            .setLazy(qryParams.lazy() && plan.canSelectBeLazy());
```

Comment:
As I understand the method can be invoke only for non MVCC mode, so I don't understand why we need check `plan.canSelectBeLazy()` here